### PR TITLE
Feature/sckan 167

### PR DESCRIPTION
**Pull request by Salam-Dalloul**

### CODE REVIEW

**Review by zsinnema — APPROVED** [review]

**Comment by zsinnema** [comment]
@Salam-Dalloul still draft?

File: backend/composer/api/filtersets.py [hunk 1]
```diff
@@ -89,6 +88,18 @@ class ConnectivityStatementFilter(django_filters.FilterSet):
         ),
     )
 
+    def filter_sentence_id(self, queryset, name, value):
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply backend/composer/api/filtersets.py]: Done :+1:

**Comment by ddelpiano** [comment]
@Salam-Dalloul @afonsobspinto is this still a draft? Thx

**Comment by afonsobspinto** [comment]
Still a draft. What's missing:
- UI for business logic errors (invalid forward connection)
- UI for No results

**Review by zsinnema — DISMISSED** [review]

File: backend/composer/api/views.py [hunk 2]
```diff
@@ -313,6 +313,19 @@ def get_queryset(self):
             return ConnectivityStatement.objects.excluding_draft()
         return super().get_queryset()
 
+    def update(self, request, *args, **kwargs):
+        # Remove 'state' from the request data
+        data = request.data.copy()
+        data.pop('state', None)
```
Comment (zsinnema) [inline backend/composer/api/views.py:319]: @afonsobspinto suggestion: would this work: pop it from the request.data and then call `.super()`

File: backend/composer/services/state_services.py [hunk 3]
```diff
@@ -79,9 +79,9 @@ def can_be_reviewed(sentence):
         # return True if the sentence can go to state to_be_reviewed
         # it should have at least one provenance (pmid, pmcid, doi) and at least one connectivity statement
         return (
-            sentence.pmid is not None
-            or sentence.pmcid is not None
-            or sentence.doi is not None
+                sentence.pmid is not None
```
Comment (zsinnema) [inline backend/composer/services/state_services.py:82]: @afonsobspinto is this correct: 2 indents?

File: backend/composer/services/state_services.py [hunk 4]
```diff
@@ -118,21 +118,21 @@ def can_be_reviewed(connectivity_statement):
...
         # origin, destination, phenotype, sex, path and species
         return (
-            connectivity_statement.origin is not None
-            and connectivity_statement.destination is not None
-            and connectivity_statement.phenotype is not None
-            and connectivity_statement.sex is not None
-            and connectivity_statement.path.count() > 0
-            and connectivity_statement.species.count() > 0
-            and connectivity_statement.provenance_set.count() > 0
+                connectivity_statement.origin is not None
```
Comment (zsinnema) [inline backend/composer/services/state_services.py:121]: @afonsobspinto same here

**Review by zsinnema — CHANGES_REQUESTED** [review]

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
Also update the export according to @ddelpiano message:
> for the export:
the relationship = forward connection;
the structure = neuron population label of the forward connection;
the identifier = NLP-ID;
the predicate = ilxtr:hasForwardConnectionPhenotype

**Review by zsinnema — APPROVED** [review]

**Review by afonsobspinto — APPROVED** [review]
Tested the export and looks good to me :+1: 

image

**Pull request merged**